### PR TITLE
Files are duplicated when added to the extra folder of the package manager if they were in a folder structure

### PR DESCRIPTION
**Pull request by pboyer**
### Purpose

When users modify the internal structure of a package directory, `PackageDirectoryBuilder` would create multiple copies of the contained files. 

I changed `PackageDirectoryBuilder` so that it does not modify the internal directory structure.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@ramramps

### CODE REVIEW

File: test/Libraries/PackageManagerTests/PackageDirectoryBuilderTests.cs [hunk 1]
```diff
@@ -180,5 +186,35 @@ public void BuildPackageDirectory_DeletesTheOriginalFiles()
         }
 
         #endregion
+
+        #region CopyFilesIntoPackageDirectory
+
+        [Test]
+        public void CopyFilesIntoPackageDirectory_DoesNotMoveFilesAlreadyWithinDirectory()
+        {
+            var files = new[] { @"C:\foo/dyf\file1.dyf", @"C:\\foo\dyf\file2.dyf", @"C:\\foo\dyf\file3.dyf",
```
Comment (ramramps) [inline test/Libraries/PackageManagerTests/PackageDirectoryBuilderTests.cs]: Why the "/" are different in the string?

File: src/DynamoPackages/PackageDirectoryBuilder.cs [hunk 2]
```diff
@@ -124,44 +124,42 @@ private void WritePackageHeader(Package package, IDirectoryInfo rootDir)
...
-                if (file == null)
-                {
-                    continue;
-                }
+            // normalize the paths to ensure correct comparison
+            var dyfDirPath = NormalizePath(dyfDir.FullName);
+            var binDirPath = NormalizePath(binDir.FullName);
+            var extraDirPath = NormalizePath(extraDir.FullName);
 
+            foreach (var file in files.Where(x => x != null))
```
Comment (ramramps) [inline src/DynamoPackages/PackageDirectoryBuilder.cs]: Why there is null check here? We should ideally pass only those files which exist. May be I am missing something here, but I think files will not be null here - but placing a null inside an IEnumerable is not good.

File: test/Libraries/PackageManagerTests/PackageDirectoryBuilderTests.cs (hunk 1 above)
Comment (pboyer) [inline test/Libraries/PackageManagerTests/PackageDirectoryBuilderTests.cs]: This is to confirm the test is resistant to differences in file paths.

File: src/DynamoPackages/PackageDirectoryBuilder.cs (hunk 2 above)
Comment (pboyer) [inline src/DynamoPackages/PackageDirectoryBuilder.cs]: This makes the API more robust.

**Pull request merged**